### PR TITLE
fix some minor bugs

### PR DESCRIPTION
**Pull request by ansyral**
use fspath rather than path as test storage cache's key because fspath would transform windows path to lower case;
fix a bug when resolving parent; and move the logic to fetchtestUtility.ts

Signed-off-by: xuzho <xuzho@microsoft.com>

### CODE REVIEW

File: extension/src/testResourceManager.ts [hunk 1]
```diff
@@ -1,18 +1,23 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 
-import { workspace, Uri } from 'vscode';
+import { commands, workspace, Uri } from 'vscode';
+import * as Commands from './commands';
+import { Logger } from './logger';
```
Comment (yaohaizh) [inline extension/src/testResourceManager.ts]: Not use of Logger?

File: extension/src/classPathManager.ts [hunk 2]
```diff
@@ -3,10 +3,14 @@
 
 import { workspace, CancellationToken, Uri } from 'vscode';
 import * as Commands from './commands';
+import { Logger } from './logger';
 
 export class ClassPathManager {
     private classPathCache = new Map<string, string[]>();
 
+    constructor(private _logger: Logger) {
```
[the remark this answers is not in the document]
Comment (ansyral) [reply extension/src/classPathManager.ts]: it is not a function but an instance method

File: extension/src/testResourceManager.ts (hunk 1 above)
Comment (ansyral) [reply extension/src/testResourceManager.ts]: e, i was implementing the feature for test explorer and found these bugs so i split the changes about the bug fixing to this pr

File: extension/src/classPathManager.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ansyral) [reply extension/src/classPathManager.ts]: #Pending# `Logger` is better to be stateless, let me refactor it later

**Pull request merged**